### DISCUSSION TITLE
Node version for GitLab CI config

### DISCUSSION THREAD

**Discussion by PhilippBaschke** [renovatebot/renovate#10163] [Request Help] [answered]
We added the following to the Renovate config to get the latest LTS version of node in our projects:

```json
"node": {
  "supportPolicy": ["lts_latest"]
},
```

There are two merge requests that update our `.nvmrc` and `.gitlab-ci.yml` file:

```diff
# .nvmrc
- v10
+ 14
```

```diff
#.gitlab-ci.yml
- image: node:12
+ image: node:16
```

Is there an extra setting that makes sure that the `.gitlab-ci.yml` node version will also use the latest LTS version instead of the latest version?

Thank you so much 🤗

**Comment by rarkins** [comment, marked as answer]
`supportPolicy` was only implemented for Travis CI. If you make sure that `node` `versioning` is used then it should know that v15 and v16 are "unstable" releases and not upgrade to them:

```json
"packageRules": 
[{
  "matchManagers": ["gitlabci"],
  "matchPackageNames": ["node"],
  "versioning": "node"
}]
```

**Reply by PhilippBaschke** [reply]
Works like a charm 🎉 

Thanks ☺️

**Comment by fgreinacher** [comment]
@rarkins I also stumbled over this behavior. Would it make sense to add something like this to `config:base`:

```json
{
      "description": "This rule ensures that the Node.js versioning logic is used for upgrading the node Docker image. See https://github.com/renovatebot/renovate/discussions/10163 for details.",
      "matchDatasources": ["docker"],
      "matchPackageNames": ["node"],
      "versioning": "node"
    }```

**Reply by viceice** [reply]
No, as we have a lot people using docker tags with suffix like `-alpine` with are currently not `node` versioning compatible

**Reply by viceice** [reply]
We need to update node versioning to treat that suffix as compatibility. I've never seen node prereleases so that behavior should be safe.

**Reply by fgreinacher** [reply]
Thanks for the quick answer, I haven't thought about the suffix, that's indeed a strong argument!

If we'd change `node` versioning to "ignore" the suffix, wouldn't that break stuff for folks that (mis)use `node` versioning for other packages (that have actual prereleases)?

**Reply by viceice** [reply]
I would not ignore suffix, i would treat it as compabillity like the docker versioning does.

I think we can ignore misusage.

**Reply by fgreinacher** [reply]
Alright, I'd take a stab at this. Should I create a separate issue?

**Reply by viceice** [reply]
yes

**Reply by fgreinacher** [reply]
👍 Here you go: #13270
